databuilder/query_model.py

```diff
@@ -43,6 +43,7 @@
 # type without specifying what that type has to be
 T = TypeVar("T")
 Numeric = TypeVar("Numeric", int, float)
+BoolOrNone = TypeVar("BoolOrNone", bool, type(None))
```

Comment:
Gah! Somehow Github lost my original comment on this, which was something like this...

I'm not sure about introducing this type. I seem to remember having a discussion earlier about this (maybe it was just BenBC and me though). But I think that the semantics of the Query Model are that every series can potentially contain NULLs, and this is just something that needs to be handled correctly by Query Engines. Can you explain a bit more about why you needed to introduce this?

Reply:
Right.  I made the change to support using `Value(value=None)` in a boolean operation:

```
Function.And(
    lhs=SelectColumn(source=SelectPatientTable(name='p', schema=None), name='b'),
    rhs=Value(value=None)
)
```

This comes from something like `pf.col & None`.

Without the change, that doesn't typecheck:

```
TypeError: Or.rhs requires 'databuilder.query_model.Series[bool]' but got 'databuilder.query_model.Series[NoneType]'
```

There's no good reason why you'd ever want to do `pf.col & None` (or `& True` or `& False`).

The reasons we do in the specs because elsewhere we test binary operations by operating on a series and a value.  But there's no good reason to do so, so we might as well not for the `bool_series` tests.